### PR TITLE
REF: split out to_datetime_with_unit, _to_datetime_with_format

### PR DESCRIPTION
**Pull request by jbrockmendel**
The idea is that to_datetime_with_format will eventually be moved into objects_to_datetime64ns, and most of _convert_listlike_datetimes will just be a call to sequence_to_dt64ns

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -517,6 +487,93 @@ def _array_strptime_with_fallback(
     return _box_as_indexlike(result, utc=utc, name=name)
 
 
+def _to_datetime_with_format(
```
Comment (jreback) [inline pandas/core/tools/datetimes.py:490]: where is this called? (now)

Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py:490]: this is basically cut/paste/de-dented from _convert_listlike_datetimes

Comment (jbrockmendel) [reply pandas/core/tools/datetimes.py:464]: woops, did the copy/paste but forgot to actually use the new function.  updated

**Comment by jreback** [comment]
doc failure looks like network; ping on green.

**Comment by jreback** [comment]
hmm can you merge master

**Pull request merged**